app/containers/CustomHeader/index.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { ReactNode } from 'react';
+import { PixelRatio, Text, View } from 'react-native';
+import { NativeStackHeaderProps } from '@react-navigation/native-stack';
+import { HeaderBackButton } from '@react-navigation/elements';
+
+import HeaderContainer from './components/HeaderContainer';
+import { isAndroid } from '../../lib/methods/helpers';
+import { useAppSelector } from '../../lib/hooks';
+import { useTheme } from '../../theme';
+import { styles } from './styles';
+
+interface IHeaderTitle {
+	headerTitle?: string | ((props: { children: string; tintColor?: string }) => ReactNode);
+}
+
+const HeaderTitle = ({ headerTitle }: IHeaderTitle) => {
+	const { colors } = useTheme();
+	if (!headerTitle) return null;
+	if (typeof headerTitle === 'string') {
+		if (isAndroid) {
+			return (
+				<Text
+					numberOfLines={1}
+					style={{
+						...styles.title,
+						paddingVertical: 10,
+						color: colors.fontTitlesLabels
+					}}>
+					{headerTitle}
+				</Text>
+			);
+		}
+		return (
+			<View style={styles.headerTitleContainer}>
+				<Text
+					numberOfLines={1}
+					style={{
+						...styles.title,
+						paddingVertical: 6,
+						color: colors.fontTitlesLabels
+					}}>
+					{headerTitle}
+				</Text>
+			</View>
+		);
+	}
+
+	return headerTitle({ children: '', tintColor: colors.fontTitlesLabels });
+};
+
+interface IHeader extends NativeStackHeaderProps {}
+
+const CustomHeader = ({ options, navigation, route }: IHeader) => {
+	const { headerLeft, headerTitle, headerRight, title } = options;
+	const { colors } = useTheme();
+	const isMasterDetail = useAppSelector(state => state.app.isMasterDetail);
+	const fontScale = PixelRatio.getFontScale();
```

Comment:
You should avoid making calls for things on render.

This hook should help you: `useWindowDimensions().fontScale`